src/StoreKeeper/WooCommerce/B2C/Helpers/Seo/YoastSeo.php

```diff
@@ -0,0 +1,200 @@
+    public static function getPostDescription(int $postId): string
+    {
+        $post = WordpressExceptionThrower::throwExceptionOnWpError(
+            get_post($postId)
+        );
+        $yoastSeoDescription = '';
+        if (Seo::isYoastActive() && !is_null($post)) {
+            $yoastSeoDescription = WordpressExceptionThrower::throwExceptionOnWpError(
+                get_post_meta($postId, '_yoast_wpseo_metadesc', true)
+            );
+
+            if (empty($yoastSeoDescription)) {
+                $yoastSeoTitles = get_option('wpseo_titles', []);
+                $postDescriptionKey = 'metadesc-'.$post->post_type;
+                $yoastSeoDescription = $yoastSeoTitles[$postDescriptionKey] ?? get_the_title($post);
+            }
+
+            return wpseo_replace_vars($yoastSeoDescription, $post);
+        }
+
+        return $yoastSeoDescription;
+    }
+
+    /**
+     * @throws WordpressException
+     */
+    public static function getPostKeywords(int $postId): string
+    {
+        $post = WordpressExceptionThrower::throwExceptionOnWpError(
+            get_post($postId)
+        );
+        $yoastSeoDescription = '';
+        if (Seo::isYoastActive() && !is_null($post)) {
+            $yoastSeoDescription = WordpressExceptionThrower::throwExceptionOnWpError(
+                get_post_meta($postId, '_yoast_wpseo_metakeywords', true)
+            );
+        }
+
+        return $yoastSeoDescription;
+    }
+
+    /**
+     * @throws WordpressException
+     */
+    public static function getCategoryTitle(int $termId): string
+    {
+        $yoastSeoTitle = '';
+        if (Seo::isYoastActive()) {
+            $termMeta = WordpressExceptionThrower::throwExceptionOnWpError(
+                get_option(self::YOAST_SEO_OPTION_KEY)
+            );
+
+            $categories = $termMeta['product_cat'];
+
+            if (isset($categories[$termId])) {
+                $category = $categories[$termId];
+                $yoastSeoTitle = $category['wpseo_title'] ?? '';
+            }
+        }
+
+        return $yoastSeoTitle;
+    }
+
+    /**
+     * @throws WordpressException
+     */
+    public static function getCategoryDescription(int $termId): string
+    {
+        $yoastSeoTitle = '';
+        if (Seo::isYoastActive()) {
+            $termMeta = WordpressExceptionThrower::throwExceptionOnWpError(
+                get_option(self::YOAST_SEO_OPTION_KEY)
+            );
+
+            $categories = $termMeta['product_cat'];
+
+            if (isset($categories[$termId])) {
+                $category = $categories[$termId];
+                $yoastSeoTitle = $category['wpseo_desc'] ?? '';
+            }
+        }
+
+        return $yoastSeoTitle;
+    }
+
+    /**
+     * @throws WordpressException
+     */
+    public static function addSeoToPost(int $postId, ?string $title = null, ?string $description = null, ?string $keywords = null): void
+    {
+        if (Seo::isYoastActive()) {
+            if (!is_null($title)) {
```

Comment:
after updating the meta we don't need to trigger anything on the Yoast plugin? Did you test it with some caching plugins?

Reply:
Tried using `W3 Total Cache`. When I directly change the meta from the database, the changes does not reflect. But once I used the `Force sync` or the `Product import`, the SEO changes were applied.

Reply:
Hmm this would mean that we need to still call some wordrpess hook.
Maybe set the meta using WP_Product class

Reply:
Used ` $product->update_meta_data`. Tested working as well even if cached.